python/dgl/nn/pytorch/explain/gnnexplainer.py

```diff
@@ -425,3 +425,441 @@ def explain_graph(self, graph, feat, **kwargs):
+                 alpha2=1.0,
+                 beta1=1.0,
+                 beta2=0.1,
+                 log=True):
+        super(HeteroGNNExplainer, self).__init__()
+        self.model = model
+        self.num_hops = num_hops
+        self.lr = lr
+        self.num_epochs = num_epochs
+        self.alpha1 = alpha1
+        self.alpha2 = alpha2
+        self.beta1 = beta1
+        self.beta2 = beta2
+        self.log = log
+
+    def _init_masks(self, graph, feat):
+        r"""Initialize learnable feature and edge mask.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            Input graph.
+        feat : dict[str, Tensor]
+            The dictionary associates input node features(values) with
+            the respective node types(keys) present in the graph.
+
+        Returns
+        -------
+        feat_masks : Dictionary of Tensor
+            Dictionary of { ntypes: features }
+            The dictionary associates the feature masks(values) with
+            the respective node types(keys).
+            Feature mask of shape :math:`(1, D_t)`, where :math:`D_t`
+            is the feature size for node type :math:`t`.
+        edge_masks : Dictionary of Tensor
+            Dictionary of { canonical_etypes: features }
+            The dictionary associates the edge masks(values) with
+            the respective canonical edge types(keys).
+            Edge mask of shape :math:`(E)`, where :math:`E` is the
+            number of edges.
+        """
+        device = graph.device
+        feat_masks = {}
+        for node_type, feature in feat.items():
+            if len(feature.size()) == 1:
+                feat_masks[node_type] = nn.Parameter(torch.zeros(1, 1, device=device))
+            else:
+                std = 0.1
+                num_nodes, feat_size = feature.size()
+                feat_masks[node_type] = nn.Parameter(torch.randn(1, feat_size, device=device) * std)
+
+        edge_masks = {}
+        for canonical_etype in graph.canonical_etypes:
+            src = canonical_etype[0]
+            num_nodes = graph.num_nodes(src)
+            num_edges = graph.num_edges(canonical_etype)
+            std = nn.init.calculate_gain('relu')
+            if num_nodes > 0:
+                std *= sqrt((2.0 / (2 * num_nodes)))
+            edge_masks[canonical_etype] = nn.Parameter(torch.randn(num_edges, device=device) * std)
+
+        return feat_masks, edge_masks
+
+    def _loss_regularize(self, loss, feat_masks, edge_masks):
+        r"""Add regularization terms to the loss.
+
+        Parameters
+        ----------
+        loss : Tensor
+            Loss value.
+        feat_mask : Dictionary of Tensor
+            Dictionary of { ntypes: features }
+            The dictionary associates the feature masks(values) with
+            the respective node types(keys).
+            Feature mask of shape :math:`(1, D)`, where :math:`D`
+            is the feature size.
+        edge_mask : Dictionary of Tensor
```

Comment:
```suggestion
        edge_mask : dict[tuple[str], Tensor]
```

Also it's better to be consistent rather than interchangeably use `edge_mask` and `edge_masks`.

Reply:
done, in the latest commit